### PR TITLE
Patch/read hdf5

### PR DESCRIPTION
**Pull request by KPHippe**
Updates: 

- Read and average embeddings into hidden dim in parallel 
- Read codon level embeddings (no average) in parallel

### CODE REVIEW

File: genslm/cmdline/run_inference.py [hunk 1]
```diff
@@ -125,6 +126,139 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
...
+    Returns
+    -------
+    np.ndarray
+        embeddings averaged into hidden_dim
+    """
+    os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
+
+    with h5py.File(h5_file_path, "r") as h5_file:
+        total_embeddings = len(h5_file["embeddings"])
+    chunk_size = total_embeddings // num_workers
```
Comment (braceal) [inline genslm/cmdline/run_inference.py:172]: Bug when total_embeddings < num_workers. Use `max(1, total_embeddings // num_workers)`

File: genslm/cmdline/run_inference.py [hunk 2]
```diff
@@ -125,6 +126,145 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
...
+    Returns
+    -------
+    np.ndarray
+        token level embeddings
+    """
+    os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
+
+    with h5py.File(h5_file_path, "r") as h5_file:
+        total_embeddings = len(h5_file["embeddings"])
+    chunk_size = total_embeddings // num_workers
```
Comment (braceal) [inline genslm/cmdline/run_inference.py:243]: Same patch needed as above: `max(1, total_embeddings // num_workers)`

**Comment by braceal** [comment]
Do we want to add optional support for reading the md5sums as well? I needed this for another application. We could return a dictionary from the process_fn’s of the form `{“embeddings”: embeddings, “md5sums”: na_hashes}` where the embeddings are either averaged or full depending on the process_fn and md5sums is only read from the h5 and returned if a Boolean argument is set to true. We could set the Boolean argument in the partial function definition like `return_md5sums=True` (this would be user specified).

**Comment by KPHippe** [comment]
smart, I can work on that. MD5 sums are good to have (and have had others ask for this as well)

**Pull request merged**